lib/index.js

```diff
@@ -403,36 +450,66 @@ exports.getFunctionUrl = async (ctx, boundaryId, functionId) => {
 
 exports.getStorageClient = (ctx) => {
     const expiry = 60 * 16; // 15+1 min to align with Lambda lifecycle plus some buffer
-    const accessToken = Jwt.sign({}, Buffer.from(ctx.configuration.fusebit_storage_key, 'base64').toString('utf8'), {
-        algorithm: 'RS256',
-        expiresIn: expiry,
-        audience: ctx.configuration.fusebit_storage_audience,
-        issuer: ctx.configuration.fusebit_storage_issuer_id,
-        subject: ctx.configuration.fusebit_storage_subject,
-        keyid: ctx.configuration.fusebit_storage_key_id,
-        header: { jwtId: Date.now().toString() },
-    });
+    let accessToken = ctx.fusebit && ctx.fusebit.functionAccessToken;
+    if (ctx.configuration.fusebit_storage_key) {
+        // Legacy mode - use pre-creaeted issuer to mint the access token
+        accessToken = Jwt.sign({}, Buffer.from(ctx.configuration.fusebit_storage_key, 'base64').toString('utf8'), {
+            algorithm: 'RS256',
+            expiresIn: expiry,
+            audience: ctx.configuration.fusebit_storage_audience,
+            issuer: ctx.configuration.fusebit_storage_issuer_id,
+            subject: ctx.configuration.fusebit_storage_subject,
+            keyid: ctx.configuration.fusebit_storage_key_id,
+            header: { jwtId: Date.now().toString() },
+        });
+    }
 
     const url = generateStorageUrl(ctx.configuration);
 
-    return {
+    const isHierarchicalStorageSupported = accessToken === ctx.fusebit.functionAccessToken;
+
+    const ensureHierarchicalStorageSupported = () => {
+        if (!isHierarchicalStorageSupported) {
+            throw new Error('Hierarchical storage is not supported in this function.');
+        }
+    }
+
+    const getUrl = (storageSubId) => `${url}${storageSubId ? ('/' + storageSubId) : ''}`;
+
+    const storageClient = {
         etag: null,
         expiration: Date.now() + expiry * 1000,
```

Comment:
@tjanczuk If you recall, we used to have this issue that if the lambda stuck around longer than expected (which we found it sometimes does), the token we generated here sometimes expired before the lambda was torn down, so storage calls would start failing. Hence the addition of this property, giving users the ability to know if the storage client is "expired" and to create a new one. What's the expiration of `functionAccessToken`, and is there a chance we end up in a similar situation if the lambda ends up sticking around for a long time?

Reply:
Discussed, the outcome is to always call `getStorageClient` and assume it's only valid as long as `ctx` is valid, which is for the duration of the request, so basically no caching the storage client.